megatron/core/models/vision/clip_vit_model.py

```diff
@@ -60,8 +60,9 @@ def __init__(
         self.img_h = img_h
         self.img_w = img_w
 
-        assert self.img_h % self.patch_dim == 0
-        assert self.img_w % self.patch_dim == 0
+        if model_subtype == "siglip":
```

Comment:
Why are we moving this assert so it's only used in the siglip case? We want it for the base clip case don't we?

Reply:
We are moving this assert to be used only in the siglip case because the model used in siglip does not satisfy this assert, while the models used in several clip cases do satisfy it. By the way, do we have any specific reason for adding this assert before? If there is no other special reason, we can consider removing it.